script.js

```diff
@@ -1,12 +1,355 @@
+  // firstOption.innerText = `Select Episode:`;
+  // firstOption.selected = true;
+  // firstOption.disabled = true;
+  // selectE.appendChild(firstOption);
+
+  // create the options for the select:
+  while (selectE.children.length > 0) {
+    selectE.children[0].remove()
+  }
+  for (const episodeData of episodeList) {
+    let option = document.createElement("option");
+    option.className = "selectOption";
+    option.value = episodeData.id;
+    option.innerText = ` ${pad(episodeData)} - ${episodeData.name
+      }`;
+    selectE.appendChild(option);
+    console.log(episodeList)
+  }
 }
 
-window.onload = setup;
+//creates DIV containing show elements (showView)
+function makeShowView(showList) {
+  //search show
+  let span = document.getElementById("count");
+  let input = document.querySelector(".search");
+  input.addEventListener("keyup", (event) => {
+
+    const searchString = event.target.value.toLowerCase();
+    const showDivs = document.querySelectorAll(".showItem");
+    let countShows = 0;
+
+
+    // Checks if each card contains the search string and if not sets display to none
+    for (let i = 0; i < showDivs.length; i++) {
+      if (!showDivs[i].innerHTML.toLowerCase().includes(searchString)) {
+        showDivs[i].style.display = "none";
+      } else {
+        showDivs[i].style.display = "block";
+        countShows++;
+      }
+    }
+    span.innerText = `Displaying ${countShows}/301 show's`;
+
+  });
+
+  //end search
+  let rootElem = document.querySelector('#root')
+  let showView = document.createElement('div');
+  showView.className = 'showView';
+  rootElem.append(showView);
+
+  for (const showData of showList) {
+    // we create show elements and add to DIV (showView)
+    let showElem = createShowElement(showData);
+    showView.append(showElem);
+  }
+
+}
+
+
+// run once when loading the page. You must call this after calling makePageLayout()
+function makeShowSelector(showList) {
+  //show select
+  let selectS = document.querySelector('.select2');
+  // let firstOption = document.createElement(`option`);
+  // selectS.appendChild(firstOption);
+  // firstOption.setAttribute(`id`, `firstShow`);
+  // firstOption.setAttribute(`value`, `select an option`);
+  // firstOption.textContent = `Select Show:`;
+  // firstOption.selected = true;
+  // firstOption.disabled = true;
+
+
+  //option for select
+  for (const showData of showList) {
+    let optionForShow = document.createElement("option");
+    optionForShow.className = "selectOptionForShow";
+    optionForShow.value = showData.id;
+    optionForShow.innerText = `${showData.name}`;
+    selectS.appendChild(optionForShow);
+
+  }
+
+  selectS.addEventListener("change", (e) => {
+    //show episode selector
+    let selectE = document.querySelector(".select");
+    selectE.style.display = '';
+
+    let showId = e.target.value;
+    // fetch data from API
+    fetch(`https://api.tvmaze.com/shows/${showId}/episodes`)
```

Comment:
Showing excellent undertsanding of a basic API fetch and how to return the data to display your episodes. There are much more you can do with API fetch/ request, but understanding these basic structure and functionality, will definitely improve your coding skills as you continue to practise.